app/src/Controller/BaseController.php

```diff
@@ -41,17 +48,21 @@
     /**
      * BaseController constructor.
      *
-     * @param $container
+     * @param \Slim\Container $container
      */
-    public function __construct($container)
+    public function __construct(Container $container)
     {
+        // fetch settings
+        $this->settings     = $container['settings'];
```

Comment:
1. Ok, just like some wordings to make code more readable
2. As to the general section/part of

```php
$this->settings     = $container['settings'];
$this->apiRenderer  = $container['apiRenderer'];
$this->validation   = $container['validation'];
$this->mailer       = $container['mailer'];
$this->mailRenderer = $container['mailRenderer'];
$this->encoder      = $container['encoder'];
```
where some objects from container are inserted inside Controller - I do not like this in general, but need some more "critical weight" in order to rework it. Say, as soon as we'll need some other provider fetched from `$container` that would be a good point when just use the whole `$container` instead